pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -133,6 +133,67 @@ func (info *reconcileInfo) publishEvent(reason, message string) {
 	info.events = append(info.events, info.host.NewEvent(reason, message))
 }
 
+// hostConfigData imaplementation of cons configuration data interface
+// Object is able to retrive data from Screts referenced in host spec
+type hostConfigData struct {
+	host         *metal3v1alpha1.BareMetalHost
+	log          logr.Logger
+	bmReconciler *ReconcileBareMetalHost
+}
+
+// Generic method for data extraction from a Secret. Function uses dataKey
+// parameter to detirmine which data to return in case secret contins multiple
+// keys
+func (hcd *hostConfigData) getSecretData(name, namespace, dataKey string) ([]byte, error) {
+	secret := &corev1.Secret{}
+	key := types.NamespacedName{
+		Name:      name,
+		Namespace: namespace,
+	}
+	if err := hcd.bmReconciler.client.Get(context.TODO(), key, secret); err != nil {
+		return nil, errors.Wrap(err,
+			"failed to fetch user data from secret reference")
```

Comment:
Now that there are multiple secrets and values, this error message should include the name of the actual secret we're trying to fetch.

Reply:
Fair enough. Fixed